bigtable-dataflow-parent/bigtable-hbase-beam/src/main/java/com/google/cloud/bigtable/beam/CloudBigtableScanConfiguration.java

```diff
@@ -276,11 +285,33 @@ public CloudBigtableScanConfiguration build() {
         if (scan == null) {
           scan = new Scan();
         }
-        ReadRowsRequest.Builder builder = Adapters.SCAN_ADAPTER.adapt(scan, readHooks);
-        request =  StaticValueProvider.of(readHooks.applyPreSendHook(builder.build()));
+        String materializedProjectId = PLACEHOLDER_PROJECT_ID;
+        String materializedInstanceId = PLACEHOLDER_INSTANCE_ID;
+        String materializedAppProfileId = PLACEHOLDER_APP_PROFILE_ID;
+
+        if(projectId != null && Strings.isNullOrEmpty(projectId.get())){
+          materializedProjectId = projectId.get();
+        }
+        if(instanceId != null && Strings.isNullOrEmpty(instanceId.get())){
+          materializedInstanceId = instanceId.get();
+        }
+        ValueProvider<String> appProfileId = additionalConfiguration.get(APP_PROFILE_ID_KEY);
+        if (additionalConfiguration.containsKey(APP_PROFILE_ID_KEY)
+            && !Strings.isNullOrEmpty(appProfileId.get())) {
+          materializedAppProfileId = appProfileId.get();
+        }
+
+        Query query = Query.create(PLACEHOLDER_TABLE_ID);
+        Adapters.SCAN_ADAPTER.adapt(scan, readHooks, query);
+        readHooks.applyPreSendHook(query);
+
+        RequestContext requestContext = RequestContext
+            .create(materializedProjectId, materializedInstanceId, materializedAppProfileId);
+
+        request = StaticValueProvider.of(query.toProto(requestContext));
```

Comment:
Hi @sduskis / @igorbernstein2 

Are there any chances of ProjectId & instanceId available at this place...(_assuming there could be that's why checking ValueProviders for projectId, instanceId & appProfileId_)?

Also recalculating ReadRowRequest at [Line#351](https://github.com/GoogleCloudPlatform/cloud-bigtable-client/pull/2011/files#diff-5836fc24866e6a80ee2b663eb55592d4R351).

Does this look good to proceed?

Reply:
We cannot proceed.  The `build()` will not necessarily have the available values in place.  Please always wrap `ValueProvider.get()` in a call to `ValueProvider.isAccessible()`

Reply:
Instead of using `Query.create(PLACEHOLDER_TABLE_ID)`, 
Shall I use `ReadRowsRequest.getDefaultInstance()`, however in both the case I would have to use Placeholders for `projectId`, `instanceId` & `appProfileId` inside `build()`.
then we can check for PlaceHolders inside RequestWithTableNameValueProvider#get and replace them with actual values.